### PR TITLE
feat: add option to use Upstash APL

### LINKED ISSUE #7: Prepare multi-tenancy
This app can work in Multi-tenant mode (single deployment, multiple Saleors)

There is preparation for that

At the moment, Upstash can be used as APL (during development)
https://github.com/saleor/saleor-app-sdk/pull/78/files#diff-02b845c068f1355ded14c02223f9762d18e10bdcbbc2cd430e474dbdb6412e4a

Later, will be replaced with Saleor Cloud APL (WIP)

### PR DESCRIPTION
**Pull request by kzuraw**
# What was done
- feat: use Upstash APL in development
- docs: add info about Upstash

Fixes #7 

## How to test it
1. Create an Upstash account and follow the [docs](https://github.com/saleor/saleor-app-slack/blob/krzysztofzuraw/issue-7/docs/upstash.md) to setup it
2. Run the application

### CODE REVIEW

**Review by lkostrowski — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -15,6 +15,10 @@ Slack app is based on App Template - you can check it [here](https://github.com/
 
 ## How to use this project
 
+### Create Upstash account
+
+You need [Upstash](https://upstash.com/) to run this application locally. Follow [How to configure Upstash for local development](docs/upstash.md) for more info.
```
Comment (krzysztofwolski) [inline README.md]: Discussed with @lkostrowski approach we want to have regarding APLs:

- add APL env variable, with default string `FileAPL`
- modify `saleor-app.ts` file to choose APL based on `APL` variable. Available choices: FileAPL, VercelAPL, UpstashAPL
- readme should contain information, that local development can be done with FileAPL (works without any dependencies). If you want to deploy or have multitenancy, use UpstashAPL.
- VercelAPL will probably be only used by the Marketplace, so don't think we have to describe it in this repository (just have it as an option in the `saleor-app.ts` file)

**Review by krzysztofwolski — CHANGES_REQUESTED** [review]

**Review by krzysztofwolski — APPROVED** [review]

File: lib/saleor-app.ts [hunk 2]
```diff
@@ -1,17 +1,12 @@
...
- * For app deployment on Vercel with Saleor CLI, use vercelAPL.
- *
- * To read more about storing auth data, read the
- * [APL documentation](https://github.com/saleor/saleor-app-sdk/blob/main/docs/apl.md)
- */
-const apl = isVercel ? new VercelAPL() : new FileAPL();
+const mapping = {
+  FileAPL: new FileAPL(),
+  VercelAPL: new VercelAPL(),
+  UpstashAPL: new UpstashAPL(),
```
Comment (lkostrowski) [inline lib/saleor-app.ts]: cc @krzysztofwolski this is different than solution in app-template. Technically its ok, but maybe we can do something to unify, at least in apps Saleor owns

Comment (krzysztofwolski) [reply lib/saleor-app.ts]: Yes, I'll update it here and klavio after merging change in the template

Comment (kzuraw) [reply lib/saleor-app.ts]: thanks for pointing this out - done 😄

**Review by lkostrowski — APPROVED** [review]

File: env.d.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+namespace NodeJS {
+  interface ProcessEnv {
+    APL_ENV: "FileAPL" | "VercelAPL" | "UpstashAPL";
```
Comment (krzysztofwolski) [inline env.d.ts]:
```suggestion
    APL: "file" | "vercel" | "upstash";
```

File: docs/upstash.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+Create or login to [Upstash](https://upstash.com/). In your [console](https://console.upstash.com/) select Redis database. On `Details` page scroll down to `REST API` section and copy `UPSTASH_REDIS_REST_URL` and `UPSTASH_REDIS_REST_TOKEN`.
+
+![](images/upstash.jpg)
+
+## Create env file
+
+Create `.env.local` file with Upstash APL env variables.
+
+```
+APL_ENV=UpstashAPL
```
Comment (krzysztofwolski) [inline docs/upstash.md]:
```suggestion
APL=upstash
```

**Review by krzysztofwolski — CHANGES_REQUESTED** [review]

**Review by krzysztofwolski — APPROVED** [review]

**Pull request merged**